appservice/src/intTest/groovy/nu/yona/server/EditGoalsTest.groovy

```diff
@@ -217,6 +215,56 @@ class EditGoalsTest extends AbstractAppServiceIntegrationTest
 		response.responseData.code == "error.goal.cannot.change.activity.category"
 	}
 
+	def 'Try invalid budget goal'()
+	{
+		given:
+		def richard = addRichard()
+		when:
+		def response = appService.addGoal(richard, BudgetGoal.createInstance(NEWS_ACT_CAT_URL, -1))
+
+		then:
+		response.status == 400
+		response.responseData.code == "error.goal.budget.invalid.max.duration.cannot.be.negative"
+	}
+
+	def 'Try invalid time zone goal'()
+	{
+		given:
+		def richard = addRichard()
+		when:
+		def noZoneResponse = appService.addGoal(richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, [].toArray()))
+		def invalidFormatResponse = appService.addGoal(richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["31:00-12:00"].toArray()))
+		def toNotBeyondFromResponse = appService.addGoal(richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["12:00-12:00"].toArray()))
+		def invalidFromHourResponse = appService.addGoal(richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["25:00-12:00"].toArray()))
+		def invalidToHourResponse = appService.addGoal(richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["11:00-25:00"].toArray()))
```

Comment:
Why is this not valid? A day can have 25 hours. See e.g. [http://infiniteundo.com/post/25326999628/falsehoods-programmers-believe-about-time]